storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java

```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
           return traces.size() == 1 && traces.get(0).size() == 2;
         });
   }
-}
\ No newline at end of file
+
+  @Test
+  public void shouldFindTracesByAnnotation() throws IOException, InterruptedException {
+
+    // todo: it is not explicit that annotations applied for tags
+    Map<String, String> annotationQuery =
+        new HashMap<String, String>() {
+          {
+            put("key_tag_a", "value_tag_a");
+          }
+        };
+
+    Span span1 =
+        Span.newBuilder()
+            .traceId("a")
+            .id("a")
+            .putTag("key_tag_a", "value_tag_a")
+            .localEndpoint(Endpoint.newBuilder().serviceName("svc_a").build())
+            .name("op_a")
+            .kind(Span.Kind.CLIENT)
+            .timestamp(Long.valueOf(TODAY + "000"))
+            .duration(10)
+            .build();
+
+    Span span2 =
+        Span.newBuilder()
+            .traceId("b")
+            .id("b")
+            .localEndpoint(Endpoint.newBuilder().serviceName("svc_b").build())
+            .putTag("key_tag_c", "value_tag_d")
+            .addAnnotation(Long.valueOf(TODAY + "000"), "annotation_b")
+            .name("op_b")
+            .kind(Span.Kind.CLIENT)
+            .timestamp(Long.valueOf(TODAY + "000"))
+            .duration(10)
+            .build();
+
+    List<Span> spans = Arrays.asList(span1, span2);
+    storage.spanConsumer().accept(spans).execute();
+
+    IntegrationTestUtils.waitUntilMinRecordsReceived(
+        testConsumerConfig, storage.spansTopic.name, 2, 10000);
+
+    // query by annotation {"key_tag_a":"value_tag_a"} = 1 trace
+    await()
+        .pollDelay(5, TimeUnit.SECONDS)
```

Comment:
Isn't the `.atMost(5, TimeUnit.SECONDS)` a better match for this case?

We are waiting for the store to pick up the change on the output topic, after we have validated that messages are published on the topic L210.
If we are validating that message is already published, then any moment, from that point, the storage could be updated.

Using `atMost` we are telling to check condition during 5 secs, and if not match, then fail.
With `pollDelay` we are waiting 5secs and then validating the condition.

It seems that the first one covers better the eventual consistency on this operation. 

WDYT?

Reply:
agree.